addon/internal-session.js

```diff
@@ -41,7 +41,10 @@ export default ObjectProxy.extend(Evented, {
 
   invalidate() {
     this._busy = true;
-    assert('Session#invalidate requires the session to be authenticated!', this.get('isAuthenticated'));
+
+    if (!this.get('isAuthenticated')) {
+      return RSVP.Promise.resolve(true);
```

Comment:
I don't think this needs to resolve with a value actually?

```js
return RSVP.Promise.resolve(true);
```

Or do we always resolve with a value here and if so what does the value actually mean?

Reply:
It only seemed odd to me to not return anything as in a similar way I would prefer to return `true` or `false` instead of `undefined` from a function that exits early. Do you think this is a weird pattern when working with Promises though and if so, why?

Reply:
I don't think there is any meaning to that return value. The idea behind the promise here is that it resolves when session invalidation was successful and rejects when it was not. What would it mean though if the returned promise resolves to `true`? Could it also resolve to something other than that? Also, as we normally return the promise returned by the authenticator's `invalidate` method that also does not resolve with a value, effectively promises that resolves with no value and promises that resolve with `true` would now both indicate a successful session invalidation.

> in a similar way I would prefer to return true or false instead of undefined from a function that exits early

I'm not sure why you're saying that. I think a function's return value is not dependent on its internal behavior (e.g. exiting early) but it's signature/semantic as seen from the outside. If there is a meaning to returning `true`/`false` or anything else, a return value makes sense but in this case we indicate a successful authentication by the fact that the returned promise resolves already and rejects when invalidation was not successful. I couldn't see how resolving with a value would add additional meaning to that.

Reply:
> Also, as we normally return the promise returned by the authenticator's invalidate method that also does not resolve with a value, [...]

I see I missed that - I agree that it would make sense to not return a value to be consistent then.

> [...] I couldn't see how resolving with a value would add additional meaning to that.

How can I check in a test case that the Promise has resolved successfully when it won't resolve with a defined value actually? I wouldn't know right now